AES.OpeningsSvc/OpeningSvc.svc.cs

```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
             DBFileManager.SetDataDirectory();
         }
 
-        public List<JobOpeningContract> GetOpenings(int StoreID)
+        public List<JobOpeningContract> GetApprovedOpenings(int StoreID)
+        {
+            return GetOpenings(StoreID, OpeningStatus.APPROVED);
+        }
+
+        public List<JobOpeningContract> GetPendingOpenings(int StoreID)
+        {
+            return GetOpenings(StoreID, OpeningStatus.PENDING_APPROVAL);
+        }
+
+        public List<JobOpeningContract> GetRejectedOpenings(int StoreID)
+        {
+            return GetOpenings(StoreID, OpeningStatus.REJECTED);
+        }
+
+        public bool RequestOpening(int StoreID, JobOpeningContract opening)
+        {
+            using (var db = new AESDbContext())
+            {
+                var store = db.Stores.Where(a => a.ID == StoreID).FirstOrDefault();
+                var job = db.Jobs.Where(j => j.JobID == opening.JobID).FirstOrDefault();
+
+                if(store == null || job  == null)
+                {
+                    return false;
+                }
+
+                var openingRequest = new JobOpening()
+                {
+                    Store = store,
+                    Job = job,
+                    Status = OpeningStatus.PENDING_APPROVAL,
+                    RequestNotes = opening.RequestNotes
+                };
+
+                db.JobOpenings.Add(openingRequest);
+                db.SaveChanges();
+            }
+            
+            return true;
+        }
+
+        public bool ApproveOpening(JobOpeningContract opening, string notes)
+        {
+            using (var db = new AESDbContext())
+            {
+                var gottenOpening = db.JobOpenings.Where(o => o.ID == opening.OpeningID).FirstOrDefault();
+                if (gottenOpening == null || gottenOpening.Status != OpeningStatus.PENDING_APPROVAL)
+                {
+                    return false;
+                }
+
+                gottenOpening.StoreManagerNotes = notes;
+                gottenOpening.Status = OpeningStatus.APPROVED;
+
+                try
+                {
+                    if (db.SaveChanges() < 1)
+                    {
+                        return false;
+                    }
+                }
+                catch (System.Exception ex)
+                {
+
+                    throw ex;
```

Comment:
Theoretically, the service should never throw an error. Plus, the catch -> throw makes this try/catch pointless. Either return false or remove the try.